kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala

```diff
@@ -107,30 +107,11 @@ object KernelCheck {
       if (d == Duration.Inf) 3896691548866406746L
       else if (d == Duration.MinusInf) 1844151880988859955L
       else if (d == Duration.Undefined) -7917359255778781894L
-      else
-        d.length * (d.unit match {
-          case DAYS         => -6307593037248227856L
-          case HOURS        => -3527447467459552709L
-          case MINUTES      => 5955657079535371609L
-          case SECONDS      => 5314272869665647192L
-          case MILLISECONDS => -2025740217814855607L
-          case MICROSECONDS => -2965853209268633779L
-          case NANOSECONDS  => 6128745701389500153L
-        })
+      else d.toNanos
     }
 
   implicit val cogenFiniteDuration: Cogen[FiniteDuration] =
-    Cogen[Long].contramap { d =>
-      d.length * (d.unit match {
-        case DAYS         => -6307593037248227856L
-        case HOURS        => -3527447467459552709L
-        case MINUTES      => 5955657079535371609L
-        case SECONDS      => 5314272869665647192L
-        case MILLISECONDS => -2025740217814855607L
-        case MICROSECONDS => -2965853209268633779L
-        case NANOSECONDS  => 6128745701389500153L
-      })
-    }
+    Cogen[Long].contramap(_.toNanos)
```

Comment:
@larsrh this is the same as the one defined in ScalaCheck currently, so maybe we should just delete it?

Reply:
Excellent point. I just checked ans ScalaCheck has all combinations of {`Arbitrary`, `Cogen`} and {`Duration`, `FiniteDuration`}. Should I delete them too?

Reply:
#3775